File: CineBox/CineBox.Model/ViewModels/Promotion.cs
using System;
namespace CineBox.Model.ViewModels
{
	public partial class Promotion
	{
        public int Id { get; set; }

        public string Code { get; set; } = null!;

        public decimal Discount { get; set; }

        public DateTime ExpirationDate { get; set; }
    }
}

File: CineBox/UI/cinebox_mobile/lib/models/Promotion/promotion.dart
import 'package:json_annotation/json_annotation.dart';

part 'promotion.g.dart';

@JsonSerializable()
class Promotion {
  int? id;
  String? code;
  double? discount;
  DateTime? expirationDate;

  Promotion();

  factory Promotion.fromJson(Map<String, dynamic> json) => _$PromotionFromJson(json);

  Map<String, dynamic> toJson() => _$PromotionToJson(this);
}